### PR TITLE
Validation

### PR DESCRIPTION
**Pull request by Rustamxon7**
### Blog App: validations, model specs, and n+1 problems

All requirements are complete:

- Validations have been added to `users`, `posts`, `likes`, and `commits` 💯
- All methods have been tested 💯
- Flash messages added successfully 💯

Feel free to review🧑🏻‍💻 my project and let me know if there are any changes to do. 🤓

### CODE REVIEW

File: spec/requests/users_spec.rb [hunk 1]
```diff
@@ -1,31 +1,17 @@
 require 'rails_helper'
 
 RSpec.describe 'Users', type: :request do
-  describe 'GET /index' do
-    it 'returns http success' do
-      get '/users/index'
-      expect(response).to have_http_status(:success)
```
Comment (arslanbisharat) [inline spec/requests/users_spec.rb:7]: - [ ] Any custom methods in the models should have a corresponding unit test.

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested, 

Hi there, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

- [ ] [Optional]  You have done with N+ 1 problem but the app solves N+1 problem when fetching all posts and their comments for a user as shown in the [wireframe](https://raw.githubusercontent.com/microverseinc/curriculum-rails/main/blog-app/images/blog_user_all_posts.png?token=ABIFLAZ4K2ZHA777536NECDBHORQS) Example in the posts#index actions students should use `@posts = @user.posts.includes(:comments)` instead of `@ppsts = @user.posts`. 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have limited number of code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+#   end
+
+#   after_save :update_post_counter
+
+#   private
+
+#   def update_post_counter
+#     author.update_attribute(:posts_counter, author.posts.count)
+#   end
+# end
```
Comment (Dipeshtwis) [inline spec/models/post_spec.rb:26]: [OPTIONAL]

- I would suggest removing the commented code to make the codebase cleaner.

**Review by Dipeshtwis — APPROVED** [review]
# Hi @Rustamxon7, 

### Status: Approved ✔️

:+1:  You have implemented the suggested changes. Well Done.

Your project is complete!  There are some optional changes I've suggested to you to make your project best, other than that nothing else to say.
Congratulations! 🎉 


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**_Check the comments under the review_**


**You may close this Pull Request and merge the changes**

  
**Keep being awesome devs and good luck with your next milestone!**
-----

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Dipesh Kumar_**

------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**